node/services/sellerService.ts

```diff
@@ -1,74 +1,117 @@
-import { settings } from '../utils'
+import { settings } from './utils'
 
 export default {
   all: async ({ ctx }: { ctx: Context }) => {
-    const sellers = await ctx.clients.sellersClient.sellers()
-    const adyenAccounts = await ctx.clients.account.all()
-    const settingsFetch = await settings(ctx)
-
-    const response = await Promise.all(
-      sellers
-        .reduce((prev, seller) => {
-          if (seller.account === ctx.vtex.account) return prev
-
-          let adyenAccount: SellerAccount | null = null
-
-          for (const account of adyenAccounts) {
-            if (
-              account.sellerId === seller.id &&
-              (!adyenAccount || account.status === 'Active')
-            ) {
-              adyenAccount = account
-              if (account.status === 'Active') {
-                break
+    const {
+      clients: { sellersClient, account: accountClient, adyenClient },
+      vtex: { logger, account: currentAccount },
+    } = ctx
+
+    try {
+      const { items: sellers } = await sellersClient.sellers()
+      const { data: adyenAccounts } = await accountClient.all()
+      const settingsFetch = await settings(ctx)
+
+      const response = await Promise.all(
+        sellers
+          .reduce((prev, seller) => {
+            if (seller.account === currentAccount) return prev
+
+            let adyenAccount: SellerAccount | null = null
+
+            for (const account of adyenAccounts) {
+              if (
+                account.sellerId === seller.id &&
+                adyenAccount?.status !== 'Active' &&
+                (!adyenAccount || account.status === 'Active')
+              ) {
+                adyenAccount = account
               }
             }
-          }
 
-          prev.push({
-            ...seller,
-            adyenAccount,
-          })
+            prev.push({
+              ...seller,
+              adyenAccount,
+            })
 
-          return prev
-        }, [] as any)
-        .map(async (seller: any) => {
-          if (seller?.adyenAccount) {
-            const accountStatus =
-              await ctx.clients.adyenClient.getAccountHolder(
+            return prev
+          }, [] as any)
+          .map(async (seller: any) => {
+            if (seller?.adyenAccount) {
+              const accountStatus = await adyenClient.getAccountHolder(
```

Comment:
Instead of doing an `await` on every seller, we can do a `promise all` for faster performance

Reply:
I thought the `Promise.all()` in [line 15](https://github.com/vtex-apps/adyen-platforms/blob/67a90f938422f3d12317dc68c85c1277bc60101b/node/services/sellerService.ts#L15) was doing this? If not, could you clarify how I'd incorporate the `Promise.all()` here?

Reply:
I think the problem is that you're using the result of the promise (`accountStatus`) in the loop, so each iteration has to wait for the promise to complete before the next iteration can start. You need to make sure to return a promise in the `.map()`, rather than returning the seller object. So you could do something like (simplified):

```
.map(async (seller: any) => {
  return adyenClient.getAccountHolder(args).then(result => {
    seller.adyenAccount.status = ...
    return seller
  })
})
```

So the `.map()` function will return an array of promises, and the result of each promise will be a seller object

Reply:
Done. When `adyenClient.getAccountHolder()` doesn't need to be run, I returned the seller object directly in the map.